senders/client_factory.go

```diff
@@ -62,7 +70,7 @@ type configuration struct {
 }
 
 func (c *configuration) Direct() bool {
```

Comment:
follow-up: I forget how we use this function, but it's starting to look a little painful